packages/client/src/app/pages/LeaderBoard/LeaderBoard.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useCallback, useEffect, useRef, useState} from "react";
+import styles from "../LeaderBoard/LeaderBoard.module.scss";
+import Avatar, {AvatarSizeEnum} from "../../components/core/Avatar/Avatar";
+
+import useLeaderBoardData from "./useLeaderBoardData";
+import Loader from "../../components/core/Loader/Loader";
+import classnames from "classnames";
+
+export interface LeaderBoardData {
+    id: number,
+    name: string,
+    surname: string,
+    login: string,
+    avatar: string,
+    score: number
+}
+
+const LeaderBoard = () => {
+
+    const [pageNum, setPageNum] = useState(0);
+    const { isLoading, error, leaderboard, hasMore } = useLeaderBoardData(pageNum);
+
+    const observer = useRef<IntersectionObserver>();
+    const lastElementRef = useCallback(
+        (node: never) => {
+            if (isLoading) return;
+            if (observer.current ) observer.current.disconnect();
+            observer.current = new IntersectionObserver((entries) => {
+                if (entries[0].isIntersecting && hasMore) {
+                    setPageNum((prev) => prev + 1);
+                }
+            });
+            if (node) observer.current.observe(node);
+        },
+        [isLoading, hasMore]
+    );
+
+    useEffect(()=>{
+        setPageNum(1);
+    }, [])
+
+    const list =
+            leaderboard.map((item, i) => {
+                const ref = leaderboard.length === i + 1 ? lastElementRef : undefined
+                const cellContent = classnames(styles['cell-default'], {
+                    [styles['cell-gold']]: item.id == 1,
+                    [styles['cell-silver']]: item.id == 2,
+                    [styles['cell-bronze']]: item.id == 3,
+                })
+
+                    return (
+                        <div className={styles.cell} key={i} ref={ref}>
+                            <div className={cellContent}>
```

Comment:
```suggestion
                            <div className={styles.cell}>
```
а в стилях прописать 
```
  cell:nth-child(1): { color: gold; }
  cell:nth-child(2): { color: silver; }
  cell:nth-child(3): { color: bronze; }
```

Тогда переменная  `cellContent` не понадобится

Reply:
Поправила